examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,399 @@
+    same_llm: bool = is_same_llm_for_all_agents()
+    llm_delegate: bool = is_llm_delegate()
+
+    # Get base LLM configuration
+    if same_llm:
+        shared_agent_config: OpenAIGPTConfig = get_base_llm_config("main LLM")
+        pro_agent_config = con_agent_config = shared_agent_config
+        feedback_agent_config: OpenAIGPTConfig = get_base_llm_config("main LLM", temperature=0.2)
+    else:
+        pro_agent_config: OpenAIGPTConfig = get_base_llm_config("for Pro Agent")
+        con_agent_config: OpenAIGPTConfig = get_base_llm_config("for Con Agent")
+        feedback_agent_config: OpenAIGPTConfig = get_base_llm_config("feedback", temperature=0.2)
+
+    system_messages: SystemMessages = load_system_messages("system_messages.json")
+    topic_name, pro_key, con_key, side = select_topic_and_setup_side(system_messages)
+
+    # Prompt for number of debate turns
+    max_turns: int = int(
+        Prompt.ask(f"How many turns should the debate continue for (Defalut is {DEFAULT_TURN_COUNT}?",
+                   default=DEFAULT_TURN_COUNT)
+    )
+
+    def create_chat_agent(
+            name: str,
+            llm_config: OpenAIGPTConfig,
+            system_message: str
+    ) -> ChatAgent:
+        """creates a ChatAgent with the given parameters.
+
+        Args:
+            name (str): The name of the agent.
+            llm_config (OpenAIGPTConfig): The LLM configuration for the agent.
+            system_message (str): The system message to guide the agent's LLM.
+
+        Returns:
+            ChatAgent: A configured ChatAgent instance.
+        """
+        return ChatAgent(
+            ChatAgentConfig(
+                llm=llm_config,
+                name=name,
+                system_message=system_message,
+            )
+        )
+
+    pro_agent = create_chat_agent("Pro",
+                                  pro_agent_config,
+                                  system_messages.messages[pro_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    con_agent = create_chat_agent("Con",
+                                  con_agent_config,
+                                  system_messages.messages[con_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    feedback_agent = create_chat_agent("Feedback",
+                                       feedback_agent_config,
+                                       FEEDBACK_AGENT_SYSTEM_MESSAGE)
+    logger.info("Pro, Con, and feedback agents created.")
+
+    # Determine user's side and assign user_agent and ai_agent based on user selection
+    agents = {"pro": (pro_agent, con_agent, "Pro", "Con"), "con": (con_agent, pro_agent, "Con", "Pro")}
+    user_agent, ai_agent, user_side, ai_side = agents[side]
+    logger.info(
+        f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+        f"LLM Delegate: {llm_delegate}"
+    )
+
+    # Clear Agents memory
+    user_agent.clear_history()
```

Comment:
I don't think `clear_history()` is needed anywhere, since we're not re-running the agents/tasks in same session

Reply:
Done